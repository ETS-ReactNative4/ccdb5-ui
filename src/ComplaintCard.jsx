import React from 'react';
import './ComplaintCard.less';

export default class ComplaintCard extends React.Component {
  render() {
    const fields = [
      'company',
      'company_public_response',
      'company_response',
      'complaint_what_happened',
      'consumer_consent_provided',
      'consumer_disputed',
      'date_received',
      'date_sent_to_company',
      'issue',
      'product',
      'state',
      'sub_issue',
      'sub_product',
      'submitted_via',
      'timely',
      'zip_code'
    ]

    return (
      <li>
          {fields
            .filter(field => typeof this.props.row[field] !== 'undefined')
            .map(field => 
              <div className="col col-12" key={field}>
                <div className="col col-3">{field}</div>
                <div className="col col-9">{this.props.row[field]}</div>
              </div>
            )
          }
      </li>
    );
  }
}