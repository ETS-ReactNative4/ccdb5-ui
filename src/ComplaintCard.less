@import (less) "base.less";

li {
  list-style-type: none;

  padding-bottom: @gutter-normal;
  border-bottom: 1px solid #ccc;
  margin-bottom: @gutter-minimum; 
}
