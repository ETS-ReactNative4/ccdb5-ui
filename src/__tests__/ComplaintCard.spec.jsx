import React from 'react';
import ComplaintCard from '../ComplaintCard';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const item = {
      company: 'ABC Corp',
      company_public_response: 'Lorem',
      company_response: 'Ipsum',
      complaint_id: '99990909',
      complaint_what_happened: 'Oh baby you. Got what I need',
      consumer_consent_provided: 'Yes',
      consumer_disputed: 'Yes',
      date_received: '2013-02-03',
      date_sent_to_company: '2013-01-01',
      issue: 'Alpha',
      product: 'Beta',
      state: 'DC',
      sub_issue: 'Gamma',
      sub_product: 'Delta',
      submitted_via: 'Email',
      timely: 'Yes',
      zip_code: '20008',
      foo: 'do not show this'
    };

    const target = renderer.create(
      <ComplaintCard key={item.complaint_id} row={item} />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).not.toContain('foo');
  });
});

