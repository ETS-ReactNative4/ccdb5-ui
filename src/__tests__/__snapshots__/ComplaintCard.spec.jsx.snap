exports[`initial state renders without crashing 1`] = `
<li>
  <div
    className="col col-12">
    <div
      className="col col-3">
      company
    </div>
    <div
      className="col col-9">
      ABC Corp
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      company_public_response
    </div>
    <div
      className="col col-9">
      Lorem
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      company_response
    </div>
    <div
      className="col col-9">
      Ipsum
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      complaint_what_happened
    </div>
    <div
      className="col col-9">
      Oh baby you. Got what I need
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      consumer_consent_provided
    </div>
    <div
      className="col col-9">
      Yes
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      consumer_disputed
    </div>
    <div
      className="col col-9">
      Yes
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      date_received
    </div>
    <div
      className="col col-9">
      2013-02-03
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      date_sent_to_company
    </div>
    <div
      className="col col-9">
      2013-01-01
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      issue
    </div>
    <div
      className="col col-9">
      Alpha
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      product
    </div>
    <div
      className="col col-9">
      Beta
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      state
    </div>
    <div
      className="col col-9">
      DC
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      sub_issue
    </div>
    <div
      className="col col-9">
      Gamma
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      sub_product
    </div>
    <div
      className="col col-9">
      Delta
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      submitted_via
    </div>
    <div
      className="col col-9">
      Email
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      timely
    </div>
    <div
      className="col col-9">
      Yes
    </div>
  </div>
  <div
    className="col col-12">
    <div
      className="col col-3">
      zip_code
    </div>
    <div
      className="col col-9">
      20008
    </div>
  </div>
</li>
`;
